lib/chef/resource.rb

```diff
@@ -1461,6 +1461,20 @@ def self.default_description(default_description = "NOT_PASSED")
       @default_description
     end
 
+    # Use a partial code fragment.  This can be used for code sharing between multiple resources.
+    #
+    # Do not wrap the code fragment in a class or module.  It also does not support the use of super
+    # to subclass any methods defined in the fragment, the methods will just be overwritten.
+    #
+    # @param partial [String] the code fragment to eval against the class
+    #
+    def self.use(partial)
+      dirname = ::File.dirname(partial)
+      basename = ::File.basename(partial, ".rb")
+      basename = basename[1..-1] if basename[0] == "_"
```

Comment:
String has `start_with?` that could be used in a few places in this PR.

Reply:
heh, yeah i was too tired yesterday to find that, then remembered what that method was called when i was trying to go to sleep...

Reply:
Given `File.exist?` and `File.exists?`... exist, I always expect the method to be `String#starts_with?` but whatever.

Reply:
I was going down the `first_char` route...